src/v/net/probes.cc

```diff
@@ -282,4 +299,187 @@ std::ostream& operator<<(std::ostream& o, const client_probe& p) {
       << ", requests_blocked_memory: " << p._requests_blocked_memory << " }";
     return o;
 }
+
+ss::future<ss::shared_ptr<ss::tls::server_credentials>>
+build_reloadable_server_credentials_with_probe(
+  config::tls_config config,
+  std::string_view service,
+  std::string_view listener_name,
+  ss::tls::reload_callback cb) {
+    auto builder = co_await config.get_credentials_builder();
+    if (!builder) {
+        co_return nullptr;
+    }
+    co_return co_await build_reloadable_credentials_with_probe<
+      ss::tls::server_credentials>(
+      std::move(*builder), service, listener_name, cb);
+}
+
+template<TLSCreds T>
+ss::future<ss::shared_ptr<T>> build_reloadable_credentials_with_probe(
+  ss::tls::credentials_builder builder,
```

Comment:
note: `ss::tls::credentials_builder` is a non-final polymorphic class, so you're subject to slicing and leaking memory here. But the documentation does say:
```
     * Intentionally "primitive", and more importantly, copyable
     * container for certificate credentials options.
```
And it was given to you in an `std::optional`, so this should be safe.

Reply:
Might be worth upstreaming a final on that class?

Reply:
Yeah, it doesn't look like these ever pass through any sort of polymorphic reference at any point, and the implementation on the ss side looks _functionally_ final. I agree it looks safe.

>worth upstreaming a final

Worth a shot. My upstream PR supporting this feature is still outstanding 🙃